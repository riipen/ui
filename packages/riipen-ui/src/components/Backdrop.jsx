import clsx from "clsx";
import PropTypes from "prop-types";
import React from "react";

class Backdrop extends React.Component {
  static propTypes = {
    /**
     * The content of the component.
     */
    children: PropTypes.node,

    /**
     * An array of custom CSS classes to apply.
     */
    classes: PropTypes.object,

    /**
     * If `true`, the backdrop is open.
     */
    open: PropTypes.bool.isRequired
  };

  static defaultProps = {
    open: false
  };

  render() {
    const { children, classes, open, ...other } = this.props;

    const className = clsx("root", open, classes);

    return (
      <React.Fragment>
        <div className={className} aria-hidden {...other}>
          {children}
        </div>
        <style jsx>{`
          .root {
            align-items: center;
            background-color: rgba(0, 0, 0, 0.5);
            bottom: 0;
            display: flex;
            justify-content: center;
            left: 0;
            position: fixed;
            right: 0;
            top: 0;
          }
        `}</style>
      </React.Fragment>
    );
  }
}

export default Backdrop;
