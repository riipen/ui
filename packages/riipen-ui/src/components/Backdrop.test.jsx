import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import Backdrop from "./Backdrop";

describe("<Backdrop>", () => {
  it("renders without errors", () => {
    let error;
    try {
      mount(<Backdrop />);
    } catch (e) {
      error = e;
    }

    expect(error).toEqual(undefined);
  });

  describe("default props", () => {
    it("sets correct default props", () => {
      const wrapper = mount(<Backdrop />);

      expect(wrapper.find("Backdrop").props().open).toEqual(false);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("children prop", () => {
    it("has given child", () => {
      const child = <h1>The children prop</h1>;

      const wrapper = mount(<Backdrop children={child} />);

      expect(
        wrapper
          .find("Backdrop")
          .children()
          .contains(child)
      ).toEqual(true);
    });
  });

  describe("classes prop", () => {
    it("applies classes to the root node of Backdrop", () => {
      const classes = ["classOne", "classTwo"];

      const wrapper = mount(<Backdrop classes={classes} />);

      expect(
        wrapper
          .find("Backdrop")
          .childAt(0)
          .hasClass(classes[0])
      ).toEqual(true);
      expect(
        wrapper
          .find("Backdrop")
          .childAt(0)
          .hasClass(classes[1])
      ).toEqual(true);
    });

    it("gives an error when classes are provided as one string", () => {
      const classes = "classOne classTwo";
      const errors = jest.spyOn(console, "error").mockImplementation();

      mount(<Backdrop classes={classes} />);

      expect(errors).toHaveBeenCalled();
    });
  });

  describe("open prop", () => {
    it("sets open prop to correct value", () => {
      const open = true;

      const wrapper = mount(<Backdrop open={open} />);

      expect(wrapper.find("Backdrop").props().open).toEqual(true);
    });
  });
});
