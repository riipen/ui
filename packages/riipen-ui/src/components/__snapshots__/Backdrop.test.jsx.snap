// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop> default props sets correct default props 1`] = `
<withClasses(Backdrop)>
  <Backdrop
    classes={
      Array [
        "riipen",
        "riipen-backdrop",
      ]
    }
    open={false}
  >
    <div
      aria-hidden={true}
      className="jsx-1867827232 root riipen riipen-backdrop"
    />
    <JSXStyle
      id="1867827232"
    />
  </Backdrop>
</withClasses(Backdrop)>
`;

exports[`<Backdrop> open prop sets open prop to correct value 1`] = `
<withClasses(Backdrop)
  open={true}
>
  <Backdrop
    classes={
      Array [
        "riipen",
        "riipen-backdrop",
      ]
    }
    open={true}
  >
    <div
      aria-hidden={true}
      className="jsx-1867827232 root riipen riipen-backdrop"
    />
    <JSXStyle
      id="1867827232"
    />
  </Backdrop>
</withClasses(Backdrop)>
`;
